sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -280,7 +283,7 @@ public com.azure.core.http.HttpClient build() {
             }
 
             AddressResolverGroup<?> resolver = nettyHttpClient.configuration().resolver();
-            if (resolver == null) {
+            if (resolver == null || setDefaultAddressResolverGroup) {
```

Comment:
Just for my knowledge, since resolver is already set when it's `null` in line 207, here `resolver == null` should always be `false`? Or could resolver be set to `null` by other methods, e.g. `HttpClient.runOn`?